node-runtime/src/encode-decode.ts

```diff
@@ -175,17 +175,17 @@ export function encode(typeTable: TypeTable, path: string, type: TypeDescription
 
         return CNPJ.strip(value);
     } else if (type === "date") {
-        if (!(value instanceof Date)) {
+        if (!(value instanceof Date) && !(typeof value === "string" && !isNaN(new Date(value).getTime()))) {
```

Comment:
This should be validated with a regex as well. You are accepting far more than only valid date (without time) strings